src/idl_parser.cpp

```diff
@@ -489,10 +489,21 @@ CheckedError Parser::Next() {
         }
 
         const auto has_sign = (c == '+') || (c == '-');
-        if (has_sign && IsIdentifierStart(*cursor_)) {
-          // '-'/'+' and following identifier - it could be a predefined
-          // constant. Return the sign in token_, see ParseSingleValue.
-          return NoError();
+        if (has_sign) {
+          // Check for +/-inf which is considered a float constant.
+          if (strncmp(cursor_, "inf", 3) == 0 &&
+              !IsIdentifierStart(cursor_[4])) {
```

Comment:
`IsIdentifierStart` does not include `0-9` which strictly speaking must be tested for here.. though very minor detail.

Reply:
Done.